guides/micronaut-dependency-injection-types/micronaut-dependency-injection-types.adoc

```diff
@@ -0,0 +1,55 @@
+common:header-top.adoc[]
+
+== Sample Project
+
+You can link:@sourceDir@.zip[download a sample application] with the code examples shown in this article.
+
+== Introduction
+
+What is a dependency? Class A has a dependency on Class B when it interacts with it in any way. For example, Class A executes a method in an instance of Class B. You typically don't instantiate class dependencies when using Micronaut Framework's dependency injection engine. Instead, you rely on the framework to provide dependency instances.
+
+NOTE: In Micronaut Framework documentation, we often use the term `Bean` and `Dependency` interchangeably.
+
+== Dependency Injection Types
+
+Micronaut Framework offers three types of dependency injection:
+
+=== Constructor Injection
+
+In constructor-based injection, Micronaut Framework provides the dependencies required for the class as arguments to the constructor.
+
+source:constructor/MessageController[]
+
+=== Field Injection
+
+With field injection, Micronaut fulfills the injection endpoints for fields annotated with `jakarta.inject.Inject`.
```

Comment:
Point out that field injection is discouraged because it increases the change of a NPE and makes it harder to understand a classes requirements

Reply:
I added a paragraph  https://github.com/micronaut-projects/micronaut-guides/pull/1071/commits/7322f48a1e9736079a917a41b07720c053dc363a. Please note that we already have a section talking about constructor injection advantages.